examples/timer32.rs

```diff
@@ -56,119 +61,112 @@ fn main() -> ! {
     let mut p2_1 = gpio.p2_1.into_output();
     let mut p2_2 = gpio.p2_2.into_output();
 
-    let mut tim32 = p.TIMER32.constrain().set_clock(_clock);
+    let tim32 = p.TIMER32.split().set_clock(_clock);
+    let mut timer0 = tim32.channel0;
+    let mut timer1 = tim32.channel1;
+    let tim32 = tim32.tim.unwrap();
 
     let mut count = Count(3, TimerUnit::Seconds);
 
     p1_0.try_toggle().unwrap();
 
     // TIM32 Channel 0 - One shot Mode
-    tim32.channel0().try_start_oneshot(count).unwrap();
+    timer0.borrow(&tim32).try_start_oneshot(count).unwrap();
```

Comment:
Seems a bit counterintuitive to keep using tim32 for every single fn of timer0 and timer1